app/src/main/kotlin/ca/rmen/android/poetassistant/settings/SettingsActivity.kt

```diff
@@ -74,13 +81,29 @@ class SettingsActivity : AppCompatActivity() {
         lateinit var mTts: Tts
 
         private var mRestartTtsOnResume = false
+        @Inject lateinit var mPrefs: SettingsPrefs
         private lateinit var mViewModel: SettingsViewModel
 
+        private val requestPermissionLauncher =
```

Comment:
These changes to support the permission requesting on api level 33+ are probably not well architectured. We have quite a bit of logic here in the activity.

I'm really in low maintenance mode right now: just doing the bare minimum to be able to continue to put updates (if needed) on the play store.

🤷‍♀️